src/test/groovy/org/gradle/android/RelocationTest.groovy

```diff
@@ -1,135 +1,375 @@
+            ':app:compileDebugAidl': FROM_CACHE,
+            ':app:compileDebugJavaWithJavac': FROM_CACHE,
+            ':app:compileDebugNdk': NO_SOURCE,
+            ':app:compileDebugRenderscript': FROM_CACHE,
+            ':app:compileDebugShaders': FROM_CACHE,
+            ':app:compileDebugSources': UP_TO_DATE,
+            ':app:compileReleaseAidl': FROM_CACHE,
+            ':app:compileReleaseJavaWithJavac': FROM_CACHE,
+            ':app:compileReleaseNdk': NO_SOURCE,
+            ':app:compileReleaseRenderscript': FROM_CACHE,
+            ':app:compileReleaseShaders': FROM_CACHE,
+            ':app:compileReleaseSources': UP_TO_DATE,
+            ':app:createDebugCompatibleScreenManifests': FROM_CACHE,
+            ':app:createReleaseCompatibleScreenManifests': FROM_CACHE,
+            ':app:generateDebugAssets': UP_TO_DATE,
+            ':app:generateDebugBuildConfig': FROM_CACHE,
+            ':app:generateDebugResources': UP_TO_DATE,
+            ':app:generateDebugResValues': FROM_CACHE,
+            ':app:generateDebugSources': SUCCESS,
+            ':app:generateReleaseAssets': UP_TO_DATE,
+            ':app:generateReleaseBuildConfig': FROM_CACHE,
+            ':app:generateReleaseResources': UP_TO_DATE,
+            ':app:generateReleaseResValues': FROM_CACHE,
+            ':app:generateReleaseSources': SUCCESS,
+            ':app:javaPreCompileDebug': FROM_CACHE,
+            ':app:javaPreCompileRelease': FROM_CACHE,
+            ':app:lintVitalRelease': SUCCESS,
+            ':app:mergeDebugAssets': FROM_CACHE,
+            ':app:mergeDebugJniLibFolders': FROM_CACHE,
+            ':app:mergeDebugResources': FROM_CACHE,
+            ':app:mergeDebugShaders': FROM_CACHE,
+            ':app:mergeReleaseAssets': FROM_CACHE,
+            ':app:mergeReleaseJniLibFolders': FROM_CACHE,
+            ':app:mergeReleaseResources': FROM_CACHE,
+            ':app:mergeReleaseShaders': FROM_CACHE,
+            ':app:packageDebug': SUCCESS,
+            ':app:packageRelease': SUCCESS,
+            ':app:preBuild': UP_TO_DATE,
+            ':app:preDebugBuild': FROM_CACHE,
+            ':app:prepareLintJar': SUCCESS,
+            ':app:preReleaseBuild': FROM_CACHE,
+            ':app:processDebugJavaRes': NO_SOURCE,
+            ':app:processDebugManifest': FROM_CACHE,
+            ':app:processDebugResources': FROM_CACHE,
+            ':app:processReleaseJavaRes': NO_SOURCE,
+            ':app:processReleaseManifest': FROM_CACHE,
+            ':app:processReleaseResources': FROM_CACHE,
+            ':app:splitsDiscoveryTaskDebug': FROM_CACHE,
+            ':app:splitsDiscoveryTaskRelease': FROM_CACHE,
+            ':app:transformClassesWithDexBuilderForDebug': SUCCESS,
+            ':app:transformClassesWithPreDexForRelease': SUCCESS,
+            // TODO For some reason this doesn't load from cache on Travis
+            ':app:transformDexArchiveWithDexMergerForDebug': { GradleVersion gradleVersion ->
```

Comment:
I think I would prefer having a method `expectedResults(androidVersion, gradleVersion)` which would then return the right `Map`. Then I could see in one place what the differences between the various Gradle versions for a fixed Android version are. Then we could get rid of the if statements checking if the value type is a `DynamicTaskOutcome`, too. WDYT?

Reply:
I'm not sure I understand. This way you can very easily see what happens with a fixed Android version. How would your solution look like?

Reply:
My solution would have a static map for each Android version and then the modifications to the map for a specific Gradle version in one place.
But it is fine as it is, too.

Reply:
Ah, mutable, brrr! :)